### PR TITLE
Track DALI_extra version in DALI

### PR DESCRIPTION
**Pull request by klecki**
GTEST binary warns about wrong sha of DALI_extra.
Refactored python test to access DALI_EXTRA_PATH in one point.
QA scripts checkout the tracked version

Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>

#### Why we need this PR?
- It fixes a problem with QA test using master of DALI_extra

#### What happened in this PR?
 - Explain solution of the problem, new feature added.
A file with sha for DALI_extra is added, it is meant to indicate the version of DALI_extra to use with this version of DALI.
 - What was changed, added, removed?
DALI_EXTRA_TAG file, error checking in GTest, .in file propagating DALI_EXTRA_TAG to C++ tests. 
 - What is most important part that reviewers should focus on?
configure_file & QA scripts.
 - Was this PR tested? How?
CI
 - Were docs and examples updated, if necessary?
NO

**JIRA TASK**: [DALI-1041]

### CODE REVIEW

**Comment by klecki** [comment]
!build

File: dali/test/dali_test_config.cc [hunk 1]
```diff
@@ -36,11 +38,33 @@ const std::string &dali_extra_path() {
...
+      auto count = fread(hash, sizeof(hash[0]), kHashLen, pipe);
+      auto ret = pclose(pipe);
+      if (ret != 0 || count != kHashLen) {
+        std::cerr << "WARNING: Could not read the sha of DALI_extra at " << *_dali_extra_path
+                  << ". Make sure DALI_EXTRA is checked out to " << DALI_EXTRA_TAG
+                  << " so the test data is in the required version." << std::endl;
+        return;
+      }
+      if (strncmp(DALI_EXTRA_TAG, hash, kHashLen) != 0) {
+        std::cerr << "Error: DALI_extra at " << *_dali_extra_path << " is not checked out to "
```
Comment (mzient) [inline dali/test/dali_test_config.cc]: Why `Error`, not `WARNING`? You let the tests run...
I'd rephrase the message to:
```
WARNING: DALI_extra (<path here>) is in different version than one required by the test. Some tests mail fail to run or produce incorrect results.
Required version: <hash>
Version found: <hash>
```

File: qa/setup_dali_extra.sh [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+# User can specify optional argument with target where to download DALI_extra
+if [ "$#" -eq 1 ]; then
+    export DALI_EXTRA_PATH=$1
+else
+    export DALI_EXTRA_PATH=/opt/dali_extra
+fi
+
 DALI_EXTRA_URL="https://github.com/NVIDIA/DALI_extra.git"
+DIR=$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )
+DALI_EXTRA_TAG_PATH="${DIR}/../DALI_EXTRA_TAG"
```
Comment (mzient) [inline qa/setup_dali_extra.sh]: You call it DALI_EXTRA_**TAG**, but in reality it's a hash. Maybe use some more generic wording that does not imply a particular Git construct - e.g. DALI_EXTRA_VERSION / DALI_EXTRA_VER.

File: dali/test/dali_test_config.cc (hunk 1 above)
Comment (klecki) [reply dali/test/dali_test_config.cc]: Done

File: qa/setup_dali_extra.sh (hunk 2 above)
Comment (klecki) [reply qa/setup_dali_extra.sh]: Done
